### PR TITLE
Use `teleport install autodiscover-node` command for Server Auto Discovery

### PR DESCRIPTION
**Pull request by marcoandredinis**
This PR changes the default agent installation to use the `oneoff` script with `teleport install autodiscover-node` instead of the current installation script.
https://goteleport.com/docs/auto-discovery/servers/

This was manually tested with the following scenarios:
- Clouds: aws, azure and GCP
- Version: OSS and ENT
- Cloud (with auto-upgrades) and self-hosted
- Distros: Ubuntu, Debian, RedHat, Amazon Linux and SLES (multiple LTS versions of each)

### CODE REVIEW

File: lib/auth/grpcserver.go [hunk 1]
```diff
@@ -4478,8 +4479,8 @@ func (g *GRPCServer) GetInstaller(ctx context.Context, req *types.ResourceReques
 	if err != nil {
 		if trace.IsNotFound(err) {
 			switch req.Name {
-			case installers.InstallerScriptName:
-				return installers.DefaultInstaller, nil
+			case types.DefaultInstallerScriptName:
```
Comment (GavinFrazar) [inline lib/auth/grpcserver.go:4482]: why the change here and below to use `types.` instead of the `installers.` var? They are equal, so maybe we should delete one of them?

**Review by GavinFrazar — APPROVED** [review]

File: lib/auth/grpcserver.go (hunk 1 above)
Comment (marcoandredinis) [reply lib/auth/grpcserver.go:4482]: There's no reason for changing things here, reverting this back to use `installers` package.

I'm not aware of the reason on why we have two constants for the same thing.

**Review by hugoShaka — APPROVED** [review]
